### PR TITLE
renamed get_mallows_loglik

### PR DESCRIPTION
**Pull request by osorensen**

### CODE REVIEW

File: tests/testthat/test-misc_cpp.R [hunk 1]
```diff
@@ -3,7 +3,7 @@ context("Testing misc C++ functions")
 test_that(
   "C++ factorial function is correct",
   expect_equal(
-    lapply(1:10, BayesMallows:::factorial),
+    lapply(1:10, factorial),
```
Comment (wleoncio) [inline tests/testthat/test-misc_cpp.R:6]: Looks fine here, but we gotta be careful not to create conflicts with `Rcpp::factorial()`. As a matter of fact, why do we need own own version of it anyway?

**Review by wleoncio — APPROVED** [review]

File: tests/testthat/test-misc_cpp.R (hunk 1 above)
Comment (osorensen) [reply tests/testthat/test-misc_cpp.R:6]: Thanks! I wasn't aware of that Rcpp function. I'll get rid of our own in a new commit to this PR.

Comment (wleoncio) [reply tests/testthat/test-misc_cpp.R:6]: Looks like `divide_by_fact()` is the only usage of `Rcpp::factorial()`in the package, and `BayesMallows:::factorial()` also has one single use on `hamming_logz()`.

Comment (osorensen) [reply tests/testthat/test-misc_cpp.R:6]: An issue with `Rcpp::factorial` is that it takes a `Rcpp::NumericVector` and returns a `Rcpp::NumericVector`, thus requiring type conversion. I'll try to use the `tgamma` function from `math.h` instead, for the cases where `BayesMallows:::factorial` was used previously. BRB.

Comment (osorensen) [reply tests/testthat/test-misc_cpp.R]: Nice if you could have a look at the latest change @wleoncio. Found a factorial function in boost which seems to work in both cases. I'll try to use boost for the binomial coefficient as well, but that comes in a later PR.

Comment (osorensen) [reply tests/testthat/test-misc_cpp.R]: Ooops, that introduced another dependency, which worked locally on my computer but not elsewhere. Reverting now.

**Comment by osorensen** [comment]
@wleoncio, I wanted to point out the following change I made to a function you have written.

Here is in this PR:
https://github.com/ocbe-uio/BayesMallows/blob/0b11b1fb868cfd668a13ccebbc4b3f12c863b878/src/misc.cpp#L143-L146

Here is the develop branch:
https://github.com/ocbe-uio/BayesMallows/blob/aa87272b611aba6bb29b0bcac5ad9c367f38fc9b/src/misc.cpp#L150-L156

The logic here is that providing `n+1` to the `tgamma` function in `math.h` gives the factorial of `n`, without having to convert to Rcpp::NumericVector. This makes the code shorter, but perhaps also less readable, since it is much easier to understand what a function named `Rcpp::factorial` does than what `tgamma` does.

I see three options:
- Keep as it was, rejecting the change in this PR.
- Use the version in this PR, with `tgamma`.
- Make a simple `double factorial(int x) { return tgamma(x + 1) };` which we can use.

Any thoughts?

**Comment by wleoncio** [comment]
> I see three options:
> 
> * Keep as it was, rejecting the change in this PR.
> 
> * Use the version in this PR, with `tgamma`.
> 
> * Make a simple `double factorial(int x) { return tgamma(x + 1) };` which we can use.
> 
> 
> Any thoughts?

Definitely the second one. Much cleaner and elegant code, perhaps all it needs to be 100% clear is to add a comment reminding that this function takes advantage of the fact that Γ(x + 1) = x!.

**Pull request merged**